std/file.d

```diff
@@ -4218,11 +4218,26 @@ private void copyImpl(scope const(char)[] f, scope const(char)[] t,
         {
             import core.stdc.wchar_ : wcslen;
             import std.conv : to;
+            import std.format : format;
 
-            if (!t)
-                t = to!(typeof(t))(toz[0 .. wcslen(toz)]);
+            // Reference resources: https://docs.microsoft.com/en-us/windows/win32/api/winbase/nf-winbase-copyfilew
+            immutable uint errno = .GetLastError();
+            const(char)[] name;
+
+            if (!f)
+                f = to!(typeof(f))(fromz[0 .. wcslen(fromz)]);
+
+            if ((errno == ERROR_PATH_NOT_FOUND) || (errno == ERROR_FILE_NOT_FOUND) ||
```

Comment:
I think `ERROR_PATH_NOT_FOUND` and `ERROR_ACCESS_DENIED` could be caused by a problem with either the source path or destination path (not existing or not accessible).

Could `ERROR_FILE_NOT_FOUND` be caused when copying over a dangling symlink?

I think it would be simpler to show both filenames in all cases when the OS API call is handling both paths at once.

Reply:
Thank you, I have already done it. Indeed, the information returned by the OS is limited, and it is not easy to try to handle it very accurately.